authors/apps/profiles/views.py

```diff
@@ -0,0 +1,84 @@
+from rest_framework import status
+from rest_framework.generics import RetrieveUpdateAPIView
+from rest_framework.permissions import IsAuthenticated, AllowAny
+from rest_framework.response import Response
+from django.shortcuts import get_object_or_404
+from .models import Profile
+from .renderers import ProfileJSONRenderer
+from .serializers import ProfileSerializer
+from .exceptions import ProfileDoesNotExist
+from rest_framework.exceptions import PermissionDenied
+import jwt
+from decouple import config, Csv
+
+
+
+class ProfileRetrieveUpdateView(RetrieveUpdateAPIView):
+
+    """
+    retrieve:
+    fetch a user's profile
+
+    update:
+    Modify a user's profile.
+
+    """
+
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ProfileJSONRenderer,) 
+    serializer_class = ProfileSerializer
+
+    def retrieve(self, request, *args, **kwargs):
+        
+        try:
+
+            profile = Profile.objects.select_related('user').get(
+                user__username=self.kwargs["username"])
+            
+            print(profile)
+
+        except Profile.DoesNotExist:
+            raise ProfileDoesNotExist
+
+            
+
+        serializer = self.serializer_class(profile)
+        print(serializer.data)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+        
+    def update(self, request, *args, **kwargs):
+        """User profile update information """
+
+        username=self.kwargs["username"]
+        """
+        get the token of logged in user, decode it to retrieve username of the user
+        compare it to username passed into this url,if there's match update user profile.
+        """
+
+        auth=request.META['HTTP_AUTHORIZATION'].split()
+        token = auth[1]
+        decoded_token = jwt.decode(token,config('SECRET_KEY'))
+        logged_user=decoded_token['username']
+        if logged_user==username:
```

Comment:
This entire logic above can be reduced to a single conditional statement
**if request.user == profile.user**